ext/standard/tests/file/bug74719.phpt

```diff
@@ -0,0 +1,54 @@
+--TEST--
+Bug #74719 Allow NULL as context
+--CREDITS--
+Alexander Holman <alexander@holman.org.uk>
+--FILE--
+<?php
+/* Prototype  : resource fopen(string filename, string mode [, bool use_include_path [, resource context]])
+ * Description: Open a file or a URL and return a file pointer 
+ * Source code: ext/standard/file.c
+ * Alias to functions: 
+ */
+
+require_once('fopen_include_path.inc');
+
+$thisTestDir =  basename(__FILE__, ".php") . ".dir";
+mkdir($thisTestDir);
+chdir($thisTestDir);
+
+$newpath = relative_include_path();
+set_include_path($newpath);
+
+$tmpfile =  basename(__FILE__, ".php") . ".tmp";
+$h = fopen($tmpfile, "w", true, NULL);
+if ($h === false) {
+    echo "Unable to open file to write\n";
+}
+else {
+    echo "open and write okay\n";
+    fwrite($h, "This is the test file");
+	fclose($h);
+	$h = fopen($tmpfile, "r", false, NULL);
+	if ($h === false) {
+	   echo "Unable to open file to read file\n";
+	}
+	else {
+	   echo "open and read okay\n";
+	   fclose($h);
+	}
+	unlink($tmpfile);
+}
+
+teardown_relative_path();
+restore_include_path();
+chdir("..");
+rmdir($thisTestDir);
+
+function runtest() {
+}
+?>
```

Comment:
This test looks a bit too complicated for what it is testing. Did you adapt it from another test file? I'm guessing that the original test was specifically dealing with include path handling, which is not really relevant here.

The test for this change should not be much more than a successful fopen() call with null context. An absolute (`__DIR__` based) path can be used to be absolutely sure that include path or cwd don't matter.

Reply:
Okay I'll update the test, and you're right I did, wasn't sure of the format so based on the other fopen tests. Do you think the change to fopen_variation4 is okay?

Reply:
It might be better to keep all the inputs and only adjust the output in variant4, but really it's fine either way. After all those cases will be tested here.

Reply:
I'll update my test and the new ones accordingly tonight

Reply:
@nikic I have updated the test changing the expected output in the variation files, with the exception of `unlink-error*` where I have removed the tests, it seemed a bit wrong to have tests meant to produce errors output `true`, if you think it should be a change instead of deletion [0eb19a8](https://github.com/php/php-src/pull/3006/commits/0eb19a81b34b2e486bf4593f8dc9a43d2523cb80) has the changed value.